src/main/java/build/buildfarm/worker/shard/CFCExecFileSystem.java

```diff
@@ -308,14 +315,63 @@ private static void checkExecErrors(Path path, List<Throwable> errors) throws Ex
     }
   }
 
+  // binary search of our sorted list of dirs
+  private static Digest findDirectoryDigest(String name, List<DirectoryNode> directories) {
+    int len = directories.size();
+    int cur = 0;
+
+    while (len != 0) {
+      DirectoryNode node = directories.get(cur + len / 2);
+      len /= 2;
+      int compare = name.compareTo(node.getName());
+      if (compare == 0) {
+        return node.getDigest();
+      }
+      if (compare > 0) {
+        // search right
+        cur += len;
+      }
+    }
+    return null;
+  }
+
+  private static boolean directoryInInputs(
```

Comment:
is "InInputs" a typo or is In referring to Index?

Reply:
It's just a crappy function name. 'is directory in input tree'. Changing it to treeContainsPath